src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -355,34 +355,26 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
             final HttpRequest httpRequest = (HttpRequest) msg;
 
             if (ProxyUtils.isChunked(httpRequest)) {
-                process(ctx, httpRequest);
+                process(ctx, httpRequest, true);
             } else {
-                if (httpRequest instanceof ReferenceCounted) {
-                    LOG.debug("Retaining reference counted message");
-                    ((ReferenceCounted) msg).retain();
-                }
+                ReferenceCountUtil.retain(httpRequest);
 
                 proxyServer.getMessageProcessingExecutor()
-                    .execute(wrapTask(() -> {
```

Comment:
we need to move `wrapTask ` into `try` block so we release the message in case an exception happens there